tools/memap.py

```diff
@@ -616,6 +619,77 @@ def generate_table(self, file_desc):
 
         return output
 
+    def generate_bars(self, file_desc, device_name=None):
+        """ Generates nice looking bars that represent the memory consumption
+
+        Returns: string containing nice looking bars
+        """
+
+        # TODO add tty detection, and width detection probably
+        WIDTH = 72
+        try:
+            # NOTE this only works on linux
+            import sys, fcntl, termios, struct
+            height, width, _, _ = struct.unpack('HHHH',
+                fcntl.ioctl(sys.stdout.fileno(), termios.TIOCGWINSZ,
+                    struct.pack('HHHH', 0, 0, 0, 0)))
+            WIDTH = min(width, WIDTH)
+        except:
+            pass
+
+        text = sum(module['size']['.text']
+            for module in self.mem_report
+            if 'size' in module)
+        data = sum(module['size']['.data']
+            for module in self.mem_report
+            if 'size' in module)
+        bss = sum(module['size']['.bss']
+            for module in self.mem_report
+            if 'size' in module)
+        rom_used = text + data
```

Comment:
Could this be:
```python
rom_used = self.mem_summary['total_flash']
```

Reply:
Why not just add them together? The math below would have problems if they differed